pybamm/models/submodels/interface/open_circuit_potential/plett_ocp.py

```diff
@@ -0,0 +1,127 @@
+                h = pybamm.PrimaryBroadcast(h, [f'{domain} particle size'])
+            else:
+                sto_surf = variables[
+                    f"{Domain} {phase_name}particle surface stoichiometry"
+                ]
+                # If variable was broadcast, take only the orphan
+                if isinstance(sto_surf, pybamm.Broadcast) and isinstance(
+                    T, pybamm.Broadcast
+                ):
+                    sto_surf = sto_surf.orphans[0]
+                    T = T.orphans[0]
+
+            ocp_surf_eq = self.phase_param.U(sto_surf, T)
+            dUdT = self.phase_param.dUdT(sto_surf)
+
+            # Bulk OCP is from the average SOC and temperature
+            sto_bulk = variables[f"{Domain} electrode {phase_name}stoichiometry"]
+            T_bulk = pybamm.xyz_average(pybamm.size_average(T))
+            ocp_bulk_eq = self.phase_param.U(sto_bulk, T_bulk)
+
+
+            c_scale = self.phase_param.c_max
+            variables[f"Total lithium in {phase} phase in {domain} electrode [mol]"] =  sto_bulk* c_scale #c_s_vol * L * A
+
+            H = self.phase_param.H(sto_surf)
+            variables[f'{Domain} electrode {phase_name}equilibrium OCP [V]'] = ocp_surf_eq
+            variables[f'{Domain} electrode {phase_name}bulk equilibrium OCP [V]'] = ocp_bulk_eq
+            variables[f'{Domain} electrode {phase_name}OCP hysteresis [V]'] = H
+
+            dU = self.phase_param.U(sto_surf,T_bulk).diff(sto_surf)
+            dQ = self.phase_param.Q(sto_surf).diff(sto_surf)
+            dQdU = dQ/dU
+            variables[f'{Domain} electrode {phase_name}differential capacity [A.s.V-1]'] = dQdU
+            variables[f'{Domain} electrode {phase_name}hysteresis state distribution'] = h
+            # H = H.orphans[0]
+            # H = pybamm.SecondaryBroadcast(H,f'{domain} electrode')
+            # H = pybamm.TertiaryBroadcast(H,f'current collector')
+            # h = pybamm.SecondaryBroadcast(h,f'current collector')
+            # ocp_surf_eq = pybamm.SecondaryBroadcast(ocp_surf_eq,[f'{domain} electrode'])
+            # ocp_bulk_eq = pybamm.PrimaryBroadcast(ocp_bulk_eq,[f'{domain} electrode'])
+            # ocp_bulk_eq = pybamm.PrimaryBroadcast(ocp_bulk_eq, [f'{domain} particle size'])
+            #! why is ocp surf not in electrode domain? why no x dimension?
+            # H = pybamm.PrimaryBroadcast(H.orphans[0],[f'current collector'])
+            # h = pybamm.PrimaryBroadcast(h,[f'{domain} electrode'])
+            H_x_av = pybamm.x_average(H)
+            h_x_av = pybamm.x_average(h)
+            if domain_options["particle size"] == "distribution":
+                if sto_surf.domains['primary'] == f"{domain} electrode":
+                    ocp_surf = ocp_surf_eq + H * h
+                else:
+                    ocp_surf = ocp_surf_eq + H_x_av * h_x_av
+            else:
+                ocp_surf = ocp_surf_eq + H * h
```

Comment:
So I have been able to get the DFN tests to pass and MPM tests to pass, although I realized I needed to x_average the h and H variables for the MPM, but not for the DFN with a psd. Is there any variable where I can have that logic execute automatically?

Reply:
There's no automatic way of doing this. If you look, for example, in the interface submodels you'll see there's a lot of logic in there for x-averaged vs not x-averaged and PSD vs no PSD.

Reply:
Something like https://github.com/pybamm-team/PyBaMM/blob/develop/pybamm/models/submodels/interface/open_circuit_potential/current_sigmoid_ocp.py#L26 might be relevant

Reply:
I think I managed to get a version working (passing all the tests) though it's not very pretty in it's current state.